src/main/repository/PromotionRepository.ts

```diff
@@ -16,7 +22,11 @@ export class PromotionRepository extends Repository<Promotion> {
       promotionQuery &&
       JSON.stringify(promotionQuery) !== JSON.stringify({})
     ) {
-      return this.applyQueryOptions(promotionQuery);
+      const promotions = await this.applyQueryOptions(promotionQuery);
+      if (promotionQuery.userId && promotions.length != 0) {
+        return this.findPromotionsUserSaved(promotionQuery.userId, promotions);
+      }
+      return promotions;
     } else {
       return this.createQueryBuilder('promotion')
```

Comment:
This class is getting ugly IMO and too complex. The next time we add more stuff to here, I'll do some refactoring and try to abstract it more